src/clients/SpokePoolClient.ts

```diff
@@ -335,10 +335,21 @@ export class SpokePoolClient extends BaseAbstractClient {
    * @returns The corresponding deposit if found, undefined otherwise.
    */
   public getDepositForFill(fill: Fill, fillFieldsToIgnore: string[] = []): DepositWithBlock | undefined {
-    const depositWithMatchingDepositId = this.depositHashes[this.getDepositHash(fill)];
-    return validateFillForDeposit(fill, depositWithMatchingDepositId, fillFieldsToIgnore)
-      ? depositWithMatchingDepositId
-      : undefined;
+    const deposit = this.depositHashes[this.getDepositHash(fill)];
+    const match = validateFillForDeposit(fill, deposit, fillFieldsToIgnore);
+    if (match.valid) {
+      return deposit;
+    }
+
+    const originChain = getNetworkName(fill.originChainId);
+    this.logger.debug({
+      at: "SpokePoolClient::getDepositForFill",
+      message: `Rejected fill for ${originChain} deposit ${fill.depositId}.`,
+      reason: match.reason,
+      deposit,
+      fill
+    });
+    return undefined;
```

Comment:
Nit: do we need to explicitly return undefined?

Reply:
*shrug* We're probably better to be explicit on return types, no?